cellarium/ml/core/module.py

```diff
@@ -383,3 +392,20 @@ def teardown(self, stage: str) -> None:
         if not self._cpu_transforms_in_module_pipeline:
             self.trainer.datamodule.collate_fn = self.trainer.datamodule.collate_fn.first_applied  # type: ignore[attr-defined]
             self._cpu_transforms_in_module_pipeline = True
+
+    def on_save_checkpoint(self, checkpoint: dict[str, Any]) -> None:
+        fit_loop = self.trainer.fit_loop
+        epoch_loop = fit_loop.epoch_loop
+        batch_progress = epoch_loop.batch_progress
+        if batch_progress.current.completed < batch_progress.current.processed:  # type: ignore[attr-defined]
```

Comment:
Handle the case when checkpointing happens at the end of the epoch.

Reply:
In this case `completed` is incremented by the PyTorch Lightning.

https://github.com/Lightning-AI/pytorch-lightning/blob/3627c5bfac704d44c0d055a2cdf6f3f9e3f9e8c1/src/lightning/pytorch/loops/fit_loop.py#L316-L317